### DISCUSSION TITLE
unplaced macros during PDN stage

### DISCUSSION THREAD

**Discussion by vijayank88** [The-OpenROAD-Project/OpenROAD#4942] [General]
@oharboe 
I came across your test case https://github.com/The-OpenROAD-Project/OpenROAD/blob/master/src/pdn/test/macros_cells_not_fixed.ok
I am also facing similar error with `macro_placement` command followed by PDN stage throws error.
```
[WARNING PDN-0234] spm_inst_0 has not been placed and fixed.
[WARNING PDN-0234] spm_inst_1 has not been placed and fixed.
[ERROR PDN-0235] Design has unplaced macros.
PDN-0235
```
How to resolve this error?
Thanks in advance...

**Comment by oharboe** [comment]
@maliberty @vijayank88 I dont recall making this test case...

**Reply by vijayank88** [reply]
Ohh sorry about that. I just seen last commit. Actually it's added by @gadfort .

**Comment by gadfort** [comment]
@vijayank88 it means you have macros in your design that have not been placed. PDN will only consider placed and locked macros and therefore requires all macros to be locked. Look at your design to find the two macros that have not been placed and locked and fix that

**Reply by oharboe** [reply]
Is this not indicative of a failed macro placement stage where the macro placement did not correctly exit with nonzero exit code?

What normal circumstances should ORFS continue after failed/incomplete macro placement?

**Reply by vijayank88** [reply]
Check this test case: https://github.com/The-OpenROAD-Project/OpenLane/issues/2116 
Earlier basic macro placement used to work fine with OpenLane. Now its throwing error.

**Reply by oharboe** [reply]
This pdn testcase is not what is needed, the testcase must be for macro placement, no?

**Reply by gadfort** [reply]
@oharboe yes you are correct, macro placement or row cutting/tap cell insertion should probably also have generated some error.

**Reply by vijayank88** [reply]
[issue_reproducible.zip](https://github.com/The-OpenROAD-Project/OpenROAD/files/14958774/issue_reproducible.zip)
here is macro placement test case. Same thing updated in OpenLane issue as well.

**Reply by vijayank88** [reply]
@gadfort 
can we continue the discussion in OpenLane issue itself?
https://github.com/The-OpenROAD-Project/OpenLane/issues/2116

**Reply by gadfort** [reply]
@vijayank88 I'm not a user of the openlane flow, so I'm not sure I can be of much help since its not an issue in any of the tools I support. You should check with the maintainers of OpenLane

**Reply by vijayank88** [reply]
Ok.. understood. I will wait for OpenLane team response
